qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/VisitSteps.java

```diff
@@ -0,0 +1,122 @@
+    public void tearDown() {
+    	deletePatient(patient);
+    	quit();
+    }
+     
+    @Given("user clicks on the find patient record app")
+    public void clickOnFindPatientRecord() {
+    	findPatientPage = homePage.goToFindPatientRecord();
+    }
+    
+    @When("user searches for the patient record")
+    public void searchForPatientRecord() {
+    	findPatientPage.enterPatient(patient.identifier);
+    	findPatientPage.waitForPageToLoad();
+    }
+    
+    @And("user clicks on the first patient record")
+    public void clickTheFirstPatientRecord() {
+    	dashboardPage = findPatientPage.clickOnFirstPatient();
+    }
+
+    @Then("the system loads the clinician facing patient dashboard")
+    public void systemLoadsClinicianFacingPatientDashboard() {
+    	assertPage(dashboardPage);
+    }
+ 
+    @When("user clicks on the start visit link")
+    public void clickOnStartVisitLink() {
+    	visitsDashboardPage = dashboardPage.startVisit();
+    }
+    
+    @Then("the system starts the patient visit")
+    public void systemStartsVisit() {
+    	assertNotNull(visitsDashboardPage.getActiveVisit());
+    }
+    
+    @When("user clicks on the add past visit link")
+    public void clickOnAddPastVisitLink() {
+    	visitsDashboardPage = dashboardPage.addPastVisit();
+    }
+    
+    @Then("the system adds the patient past visit")
+    public void systemAddsPastVisit() {
+    	assertThat(visitsDashboardPage.getVisitList().get(0).getAttribute("class"), is(not("no-results")));
+    }
+    
+    @When("user clicks on the recent visits link in the recent visits section")
+    public void clickOnRecentVisitsLink() {
+    	visitsDashboardPage = dashboardPage.goToRecentVisits();
+    	visitsDashboardPage.waitForPage();
+    }
+    
+    @And("user clicks on the actions button")
+    public void clickActionsButton() {
+    	visitsDashboardPage.clickOnActions();
+    }
+    
+    @And("user clicks on the merge visits link")
+    public void clickOnMergeVisitsLink() {
+    	mergeVisitsPage = visitsDashboardPage.clickOnMergeVisits();
+    }
+    
+    @And("user selects the available visits to be merged")
+    public void selectFirstVisit() {
+    	mergeVisitsPage.checkFirstVisit();
+    	mergeVisitsPage.checkSecondVisit();
+    }
+    
+    @And("user clicks on the merge selected visits button")
+    public void clickOnMergeVisitsButton() {
+    	mergeVisitsPage.clickOnMergeSelecetdVisits();
+    }
+    
+    @Then("the system merges the visits successfully")
+    public void systemMergesVisitsSuccessfully() {
+    	assertThat(mergeVisitsPage.getAllVisit().size(), is(1));
+    }
+      
+    @When("user clicks on the end visit button")
+    public void clickOnTheEndVisitButton() {
+    	visitsDashboardPage.endVisit();
+    	visitsDashboardPage.waitForPageToLoad();
+    }
+    
+    @Then("the system ends the visit")
+    public void systemEndsVisit() {
+    	assertNull(visitsDashboardPage.getActiveVisit());
```

Comment:
we can assert the text visible on the page when the system ends the patient visit.